Makefile

```diff
@@ -1,57 +1,66 @@
-GOMOBILE=gomobile
-GOBIND=$(GOMOBILE) bind
-XGOCMD=xgo
-BUILDDIR=$(shell pwd)/build
+BUILDDIR=$(CURDIR)/build
+GOBIN=$(CURDIR)/bin
+
+GOMOBILE=$(GOBIN)/gomobile
 IMPORT_PATH=github.com/Jigsaw-Code/outline-go-tun2socks
-ELECTRON_PATH=$(IMPORT_PATH)/outline/electron
-LDFLAGS='-s -w'
+
+.PHONY: android-outline android-intra linux apple windows clean
+
+all: android-outline android-intra linux apple windows
+
 ANDROID_LDFLAGS='-w' # Don't strip Android debug symbols so we can upload them to crash reporting tools.
+ANDROID_BUILDDIR=$(BUILDDIR)/android
+ANDROID_ARTIFACT=$(ANDROID_BUILDDIR)/tun2socks.aar
+ANDROID_BUILD_CMD=$(GOMOBILE) bind -a -ldflags $(ANDROID_LDFLAGS) -target=android -tags android -work -o $(ANDROID_ARTIFACT)
+
+android-outline: $(GOMOBILE)
+	$(ANDROID_BUILD_CMD) $(IMPORT_PATH)/outline/android $(IMPORT_PATH)/outline/shadowsocks
```

Comment:
Why does this call `gomobile` twice?  Is it for `gomobile init`?  If so, can we make that explicit?

Reply:
Only the first time. The `$(GOMOBILE)` in the dependency makes it explicit. What do you mean with making it explicit?

Reply:
I would make a new target like

```make
gomobile-init:
        $(GOMOBILE) init
```

and then depend on that.

Reply:
The advantage of tying it to the gomobile install is that I init it only once. With your proposal it will call init every time.

Reply:
OK, I understand.  I would like to structure this a bit differently but I will send a followup PR for that.